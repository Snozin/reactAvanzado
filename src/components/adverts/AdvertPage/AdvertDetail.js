import React from 'react';
import T from 'prop-types';

import { ConfirmationButton } from '../../common';
import placeholder from '../../../assets/images/placeholder.png';
import { advert } from '../propTypes';

function AdvertDetail({ name, sale, price, tags, photo, onDelete }) {
  return (
    <div>
      <p>{name}</p>
      <p>{sale ? 'Sell' : 'Buy'}</p>
      <p>{tags.join(', ')}</p>
      <p>{price}</p>
      <img
        src={photo || placeholder}
        alt={name}
        width="200"
        height="200"
        style={{ objectFit: 'contain' }}
      />
      <ConfirmationButton confirmation="Are you sure?" onConfirm={onDelete}>
        Delete
      </ConfirmationButton>
    </div>
  );
}

AdvertDetail.propTypes = {
  ...advert,
  photo: T.string,
  onDelete: T.func.isRequired,
};

AdvertDetail.defaultProps = {
  photo: null,
};

export default AdvertDetail;
