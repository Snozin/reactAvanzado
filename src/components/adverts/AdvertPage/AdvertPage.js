import { useEffect } from 'react'
import { Redirect, useParams, useHistory } from 'react-router-dom'

import Layout from '../../layout'
import AdvertDetail from './AdvertDetail'

import { useSelector, useDispatch } from 'react-redux'
import { getUIState, getSingleAdvert } from '../../../store/selectors'
import { getAdvertById, deleteAdvert } from '../../../store/actions'

function AdvertPage() {
  const dispatch = useDispatch()
  const { advertId } = useParams()
  const { error } = useSelector(getUIState)
  const advert = useSelector(getSingleAdvert)
  const history = useHistory()

  useEffect(() => {
    dispatch(getAdvertById(advertId))
  }, [dispatch, advertId])

  const handleDelete = () => {
    dispatch(deleteAdvert(advertId, history))
  }

  if (error?.statusCode === 401) {
    return <Redirect to="/login" />
  }

  if (error?.statusCode === 404) {
    return <Redirect to="/404" />
  }

  return (
    <Layout>
      {advert && <AdvertDetail {...advert} onDelete={handleDelete} />}
    </Layout>
  )
}

export default AdvertPage
