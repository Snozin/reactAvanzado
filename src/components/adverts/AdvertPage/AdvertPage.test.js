import AdvertPage from './AdvertPage'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

describe('AdvertPage', () => {
  test('snapshot test', () => {
    const store = {
      getState: () => ({
        ui: {},
        adverts: {
          tags: [],
          getById: {
            name: '',
            sale: true,
            price: 0,
            tags: [],
          },
        },
      }),
      subscribe: () => {},
      dispatch: () => {},
    }

    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <AdvertPage />
          </BrowserRouter>
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
