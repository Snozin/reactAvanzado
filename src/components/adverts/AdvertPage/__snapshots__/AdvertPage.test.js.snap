// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvertPage snapshot test 1`] = `
Array [
  <header>
    Nodepop React
    <nav>
      <ul>
        <li>
          <a
            aria-current={null}
            href="/adverts"
            onClick={[Function]}
          >
            Nodepop
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/adverts/new"
            onClick={[Function]}
          >
            New advert
          </a>
        </li>
      </ul>
    </nav>
    <a
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </header>,
  <main>
    <div>
      <p>
        
      </p>
      <p>
        Sell
      </p>
      <p>
        
      </p>
      <p>
        0
      </p>
      <img
        alt=""
        height="200"
        src="placeholder.png"
        style={
          Object {
            "objectFit": "contain",
          }
        }
        width="200"
      />
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </main>,
  <footer>
    @Keepcoding 2021
  </footer>,
]
`;
